src/main/java/org/terasology/combatSystem/weaponFeatures/events/PrimaryAttackEvent.java

```diff
@@ -27,6 +28,14 @@ public PrimaryAttackEvent(ActivateEvent info){
         hitPosition = info.getHitPosition();
         hitNormal = info.getHitNormal();
         activationId = info.getActivationId();
+        EntityRef entity1 = getInstigator();
```

Comment:
Instead of using the accessor method `getInstigator()` you can directly use `instigator` as defined in line 24. 
Also, I encourage you to avoid variable names like `entity1`, `action2`, etc. if you can find better names. Here, I would go for `attacker` and `attackee`.
```suggestion
        EntityRef attacker = instigator;
```

Reply:
Furthermore, in the scope of this constructor, you already have both `instigator` and `target` at hand, so no need to redefine them with different names.
```suggestion
```